astropy/io/votable/converters.py

```diff
@@ -389,6 +404,11 @@ def _binoutput_var(self, value, mask):
                 value = value.encode("ascii")
             except ValueError:
                 vo_raise(E24, (value, self.field_name))
+
+        if self.arraysize != "*" and len(value) > self.arraysize:
+            vo_warn(W46, ("char", self.arraysize), None, None)
+            value = value[: self.arraysize]
```

Comment:
is this a reasonable behaviour when `len(value) > self.arraysize`, should this maybe raise an exception rather than just a warning?

Reply:
That's an interesting point, are you suggesting we raise an exception in the output methods or the parse methods as well? I've gone with raising warning to keep consistent with the default Char parse method and generally in favour of being more forgiving for users, but I'm happy to also raise an exception if you think this is more appropriate here. If so do you want me to do the same for the parsing method?

Reply:
Yes, but neither the Char parser no here in e.g. L385 has any value loss, while in this method you chop the return at arraysize. 
Maybe keep it a warning but don't chop?

Reply:
Thinking about this I think for consistency it may be best to truncate in the parser and the output method.
Providing a warning without truncating has potential to cause unexpected compatibility issues later on for users, for example if a user pre-allocates fixed buffers based on declared length. Also it feels wrong to generate a Table that will not conform with the schema.

If we didn't want to truncate, I'd also consider increasing the arraysize along with a warning, but that may also lead to unexpected issues.

Any thoughts on truncating both at parse and output? Otherwise I can also just not chop, the potential for the error occurring through some round-trip where on the other end they assume data that conforms with the bound is probably rare.

Reply:
I would lean towards not truncating in here, so it will be consistent (if wrong) with the rest of the methods, and thus can be easily be backported. Then this PR can go in now without further discussion.

But then opening a bug report issue where to ask for input, and possibly changing the behaviour, either by truncating or changing the schema. The former feels more correct to me, but I would like to see the input from others, too. Then potentially a fix/refactor; that is consistent with the Char methods, too, can go in the next feature release as a breaking change.

Reply:
Ok sounds good, I've removed the truncation for the output method. I've also added a few more test methods since I noticed codecov was complaining about the coverage. Let me know if you think anything else needs to be changed.

Reply:
Thanks!

Not sure why codecov is still complaining, I would suggest to ignore it for now and go ahead with the merge once CI is all green. 

Would you mind opening an issue outlining the concerns you mentioned in https://github.com/astropy/astropy/pull/18105#discussion_r2100985240 ?

Reply:
By the way I noticed the CI failed during the link check here:

`(api/astropy.io.votable.validator.make_validation_report: line   23) broken    http://www.starlink.ac.uk/stilts - HTTPConnectionPool(host='www.starlink.ac.uk', port=80): Max retries exceeded with url: /stilts (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f33be2d7350>, 'Connection to www.starlink.ac.uk timed out. (connect timeout=180)'))`

Is this a known issue? I noticed that it is using an older URL for the starlink docs:
https://github.com/astropy/astropy/blob/16a94b4d4f9f5070115dac7f52a3a1bdfbdca2cb/docs/conf.py#L612

This should be the new URL I think:
https://www.star.bristol.ac.uk/mbt/stilts/

Reply:
Unrelated, so ignore it or open a separate PR.

(actually I'm not sure why we require the extra CIs for this PR, but at this point I'm just waiting for all the jobs on the exotic arc to finish)